velox/dwio/common/SelectiveRepeatedColumnReader.cpp

```diff
@@ -63,7 +63,7 @@ void prepareResult(
           result->encoding() == VectorEncoding::Simple::ARRAY) ||
          (type->kind() == TypeKind::MAP &&
           result->encoding() == VectorEncoding::Simple::MAP)) &&
-        result.unique())) {
+        result.use_count() == 1)) {
```

Comment:
I am wondering whether we should even rely on use_count as its unreliable , see for e.g : https://stackoverflow.com/a/77286276 .

Reply:
Correct, it is unreliable. This change is a 1-1 replacement. Using `unique` was already unreliable and I guess that is why it is being removed. It will need some change eventually.

Reply:
Understood , I will create another issue to remove the use of shared_ptr::use_count == 1 from the code base as its unreliable and not meant for use across threads.

Reply:
Ahh, was about to create this issue :)

Reply:
https://github.com/facebookincubator/velox/issues/11344